core/src/main/scala/fs2/job/JobManager.scala

```diff
@@ -134,11 +141,12 @@ final class JobManager[F[_]: Concurrent: Timer, I, N] private (
     Concurrent[F].delay(Option(meta.get(id)).map(_.status))
 
   private def shutdown: F[Unit] =
-    Concurrent[F].delay(meta.clear()) >> notificationsQ.enqueue1(None)
+    Concurrent[F].delay(meta.clear()) >> notificationsQ.enqueue1(None) >> eventQ.enqueue1(None)
 
   private[this] def managementMachinery[A](
       id: I,
       in: Stream[F, A],
+      startingTime: OffsetDateTime,
```

Comment:
I might be misunderstanding this, but this method is called when we enqueue a job. So we don't yet know how long it will take to finish. We need to wait for the job to finish or fail to calculate a duration.

Reply:
"Duration" is something of a misnomer. `FiniteDuration` is literally just a unit of time. So in this case, `OffsetDateTime` is equal to `FiniteDuration`, where the duration is the amount of time since epoch.

Reply:
I see. Should I also change [`Event`](https://github.com/slamdata/fs2-job/pull/4/files#diff-823af7e95bed438f15c9a82194c38e5e) to use epochs too? I'm not opposed but I don't know if there's any benefit to using epochs over `OffsetDateTime`.

Reply:
I would. The API is a lot better and it integrates better with other Scala things, such as `Stream.sleep` and such.

Reply:
Will do!

Reply:
I have implemented this. So this should be ready for another look.